common/crypto/platform/openssl/crypto.cpp

```diff
@@ -66,6 +66,36 @@ auto bio_free_all_func = [](BIO *bio) {
 	}
 };
 
+// NOTE: GetOpenSSLErrorMessage should be called upon all OpenSSL errors, as
+// the errors are queued, and if not harvested, the FIFO structure of the
+// queue will mean that if you just get one, you might actually get the wrong
+// one.
+string GetOpenSSLErrorMessage() {
+	const auto sysErrorCode = errno;
+	auto sslErrorCode = ERR_get_error();
+
+	std::string errorDescription {};
+	if (sslErrorCode != 0) {
+		while (sslErrorCode != 0) {
+			if (!errorDescription.empty()) {
+				errorDescription += '\n';
+			}
+			errorDescription += ERR_error_string(sslErrorCode, nullptr);
+			sslErrorCode = ERR_get_error();
+		}
+	}
```

Comment:
Now that the second condition is out of this `if` block, the code could be simplified with:
```suggestion
	while (sslErrorCode != 0) {
		if (!errorDescription.empty()) {
			errorDescription += '\n';
		}
		errorDescription += ERR_error_string(sslErrorCode, nullptr);
		sslErrorCode = ERR_get_error();
	}
```